File: Locust/TesteAlbum.py
# -*- coding: utf-8 -*-
from locust import HttpLocust, TaskSet, task
import random

class TesteAlbum(TaskSet):
    userID = random.randint(1,5001)
    @task
    def create(self):
        response = self.client.post("/albums", {"title": "Album de fotos do Bicampeonato do clube Atlético Mineiro","userId": str(self.userID)})
        self.client.delete("/albums/" + str(response.json()['id']))
      
class PoCLocust(HttpLocust):
    task_set = TesteAlbum
    host = "https://jsonplaceholder.typicode.com"
    sock = None
def __init__(self):
        super(PoCLocust, self).__init__()

File: Gatling/TesteAlbum.scala
package albums

import scala.concurrent.duration._

import io.gatling.core.Predef._
import io.gatling.http.Predef._
import io.gatling.jdbc.Predef._
import scala.util.parsing.json.JSONObject

class Album extends Simulation {

val feeder = Iterator.continually(Map(("userIdGerado", scala.util.Random.nextInt(5000))))

  val headers = Map(
    "Content-Type" -> "application/json"
  )

  val httpConfiguration = http.baseUrl("https://jsonplaceholder.typicode.com/")
    .headers(headers)

  val postPassengerCenarioCarga = scenario("POC Gatling")
    .feed(feeder)
    .exec(
      http("Create Albums")
        .post("albums")
        .body(StringBody(
          JSONObject.apply(
            Map(
                "userId"->"${userIdGerado}",
                "title"-> "Album de fotos do Bicampeonato do clube Atlético Mineiro"
               )
          ).toString()
        ))
        .check(jsonPath("$.id").saveAs("id"))
    )
    .exec(
      http("Delete Albums")
        .delete("albums/${id}")
         )
  
  setUp(
    postPassengerCenarioCarga.inject(
      rampUsers(50) during (1 minutes)
    )
  ).protocols(httpConfiguration)
    .maxDuration(2 minutes)
}